src/main/java/io/confluent/connect/hdfs/HdfsSinkConnectorConfig.java

```diff
@@ -399,6 +463,143 @@ public Configuration getHadoopConfiguration() {
     return map;
   }
 
+  /**
+   * Performs all substitutions on `logs.dir` and calculates the final directory for a topic
+   *
+   * @param topic - String - the topic whose directory to find
+   * @return String - the directory name and path
+   */
+  public String getLogsDirFromTopic(String topic) {
+    return getDirFromTopic(getString(LOGS_DIR_CONFIG), topic, logDirGroupsMaxIndex);
+  }
+
+  /**
+   * Performs all substitutions on `topics.dir` and calculates the final directory for a topic
+   *
+   * @param topic - String - the topic whose directory to find
+   * @return String - the directory name and path
+   */
+  public String getTopicsDirFromTopic(String topic) {
+    return getDirFromTopic(getString(TOPICS_DIR_CONFIG), topic, topicDirGroupsMaxIndex);
+  }
+
+  /**
+   * Performs all substitutions and calculates the final directory for a topic
+   *
+   * @param dir - String - the directory to perform substitutions on
+   * @param topic - String - the topic whose directory to find
+   * @return String - the directory name and path
+   */
+  private String getDirFromTopic(String dir, String topic, int maxIndexParts) {
+    dir = dir.replace("${topic}", topic);
+
+    // only if configured
+    if (topixRegexCaptureGroup != null) {
+
+      // find all of the captured groups by the regex
+      Matcher matcher = topixRegexCaptureGroup.matcher(topic);
+      if (!matcher.matches()) {
+        throw new ConfigException(
+            String.format(
+                "Topic %s does not fully match the specified regex %s in %s",
+                topic,
+                topixRegexCaptureGroup.pattern(),
+                TOPIC_REGEX_CAPTURE_GROUP_CONFIG
+            )
+        );
+      }
+
+      // make sure that all references to captured groups actually exist
+      if (maxIndexParts > 0 && maxIndexParts > matcher.groupCount()) {
+        throw new ConfigException(
+            String.format(
+                "Topic %s must have at least %d capture groups using regex pattern %s, "
+                    + "but actually had %d capture groups.",
+                topic,
+                maxIndexParts,
+                topixRegexCaptureGroup.pattern(),
+                matcher.groupCount()
+            )
+        );
+      }
+
+      for (int index = 1; index < matcher.groupCount() + 1; index++) {
```

Comment:
Might as well give them all the groups
```suggestion
      for (int index = 0; index < matcher.groupCount() + 1; index++) {
```

Reply:
initially said I wouldn't support but maybe I should